migrator/migration-steps.spec.js

```diff
@@ -335,4 +347,194 @@ describe('migration steps', () => {
+    const matchFound = false
+    const originalFileContents = `/* global $ */
+
+// Warn about using the kit in production
+if (window.console && window.console.info) {
+  window.console.info('GOV.UK Prototype Kit - do not use for production')
+}
+
+$(document).ready(function () {
+  window.GOVUKFrontend.initAll()
+  $('.my-button').click(() => console.log('clicked'))
+})
+`
+    const expectedFileContents = `/* global $ */
+
+//
+// For guidance on how to add JavaScript see:
+// https://prototype-kit.service.gov.uk/docs/adding-css-javascript-and-images
+//
+
+
+$(document).ready(function () {
+  $('.my-button').click(() => console.log('clicked'))
+})
+`
+
+    fsp.readFile.mockReturnValue(originalFileContents) // For the second call
+
+    // mock call upgradeIfPossible method (get this working first)
+    const result = await upgradeIfPossible(applicationJsFile, matchFound)
+    expect(result).toBeTruthy()
+    const expectedFileName = path.join(projectDir, applicationJsFile)
+
+    expect(fsp.readFile).toHaveBeenCalledTimes(1)
+    expect(fsp.readFile).toHaveBeenCalledWith(expectedFileName, 'utf8')
+
+    expect(fsp.writeFile).toHaveBeenCalledTimes(1)
+    const [actualFileName, actualFileContent] = fsp.writeFile.mock.calls[0]
+    expect(actualFileName).toEqual(expectedFileName)
+    expect(actualFileContent).toEqual(expectedFileContents)
+
+    expect(reporter.addReporter).toHaveBeenCalledTimes(1)
+    expect(reporter.addReporter).toHaveBeenCalledWith(`Update ${applicationJsFile}`)
+
+    expect(mockReporter).toHaveBeenCalledTimes(1)
+    expect(mockReporter).toHaveBeenCalledWith(true)
+  })
+
+  it('upgrade filters file if possible', async () => {
+    const filtersJsFile = path.join('app', 'assets', 'javascripts', 'filters.js')
+    const matchFound = false
+    const originalFileContents = `module.exports = function (env) {
+  var filters = {}
+
+  /* ------------------------------------------------------------------
+    add your methods to the filters obj below this comment block:
+    @example:
+
+    filters.sayHi = function(name) {
+        return 'Hi ' + name + '!'
+    }
+    
+    For more on filters and how to write them see the Nunjucks
+    documentation.
+
+  ------------------------------------------------------------------ */
+
+  function getData(){
+    var dummyData = require('./data/data.js')
+    return dummyData;
+  }
+
+  filters.toID = function (str) {
+    return str.replaceAll(" ", "-").toLowerCase();
+  }
+
+  /* ------------------------------------------------------------------
+    keep the following line to return your filters to the app
+  ------------------------------------------------------------------ */
+  return filters
+}
+`
+    const starterFileContents = `const govukPrototypeKit = require('govuk-prototype-kit')
+const addFilter = govukPrototypeKit.views.addFilter
+`
+    const expectedFileContents = `const govukPrototypeKit = require('govuk-prototype-kit')
```

Comment:
I think we should add the comment that's in the starter kit.

I'd also like to remove the big old comment if it's unchanged - technically it still works but I think the user is better off having the new comment instead of the old comment.

Reply:
I've just spotted that there's not a comment in the starter kit, I think we should remove the old one anyway.